### ISSUE TITLE
Update Kubernetes Compatibility doc section

### ISSUE THREAD

**Issue by markpollack** [spring-cloud/spring-cloud-dataflow#2779]
The doc should be udpated to list just Server 2.0 and which versions of k8s are supported.  The last column is 1.10.x, which is outdated.  Perhaps at this point we just do latest - 3 versions?  Cross check with PKS and underlying k8s versions.

Could also invert the paragraph order, start with the summary and then list resources for more information on the deployers... seems like it isn't even really necessary.

**Comment by chrisjs** [comment]
looking at: http://docs.spring.io/spring-cloud-dataflow/docs/2.0.0.BUILD-SNAPSHOT/reference/htmlsingle/#_kubernetes_compatibility the chart is current with what we have deployed

* on GKE we have 1.9 - 1.11
* on PKS the current env is 1.11, we don't have a cluster for each like GKE (we could at the increase of AT time)

PKS 1.3 was released on 1/16 which brings in kubernetes 1.12. GKE has not yet GA'd 1.12

so.. in regards to versions, we could:

1) stick with the existing documentation stating we support 1.9 - 1.11 
2) drop GKE 1.9 and upgrade PKS to 1.12 which would then leave us with a compat chart of 1.10 - 1.12. this adds a little more work for tile/cluster/some manual updates/etc, plan re-working, etc but not too terrible.

In either case, we can drop the line items for 1.4.x - 1.7.x a mentioned and only have 2.0 server

Adding the deployer versioning to the table was a request from @sabbyanandan -- I'm in favor of dropping the deployer version references as its another thing to remember to update.
